js/lib/higlass-jupyter.js

```diff
@@ -20,40 +20,165 @@ var HiGlassDisplayModel = widgets.DOMWidgetModel.extend({
 
 // Custom View. Renders the widget model.
 var HiGlassDisplayView = widgets.DOMWidgetView.extend({
-  render: function() {
-    var viewConfig = this.model.get('viewconf');
-    var height = this.model.get('height');
-    var hgOptions = this.model.get('hg_options');
+  render: function render() {
+    this.height = this.model.get('height');
+    this.viewConfig = this.model.get('viewconf');
+    this.authToken = this.model.get('auth_token');
+    this.bounded = this.model.get('bounded');
+    this.defaultTrackOptions = this.model.get('default_track_options');
+    this.darkMode = this.model.get('dark_mode');
+    this.renderer = this.model.get('renderer');
+    this.selectMode = this.model.get('select_mode');
+    this.selectionOnAlt = this.model.get('selection_on_alt');
+    this.options = this.model.get('options');
 
-    var borderColor = hgOptions.isDarkTheme ? '#333333' : '#dddddd'
+    this.hgContainer = document.createElement('div');
+    this.hgDisplay = document.createElement('div');
+    this.hgDisplay.style.border = this.options.theme === 'dark'
+      ? '#333333' : '#dddddd';
+    this.hgDisplay.style.borderRadius = '2px';
 
-    this.hgcontainer = document.createElement('div');
-    this.hgdisplay = document.createElement('div');
-    this.hgdisplay.style.border = '1px solid #dddddd';
-    this.hgdisplay.style.borderRadius = '2px';
+    this.hgContainer.appendChild(this.hgDisplay);
+    this.el.appendChild(this.hgContainer);
 
-    this.hgcontainer.appendChild(this.hgdisplay);
-    this.el.appendChild(this.hgcontainer);
+    this.hg = hglib.viewer(this.hgDisplay, this.viewConfig, this.getOptions());
 
-    if (height) {
-      this.hgdisplay.style.height = height + 'px';
+    window.hgApi = this.hg;
 
+    // Listen to events from the JavaScript world
+    this.hg.on('location', this.setLocation.bind(this));
```

Comment:
Is this a per-view property? If you have two views and change the location of one, what does the notification look like?

Reply:
These should be widget-wide, no?

Reply:
I fixed this issue by supporting multiple views as well. When there is only a single view `widget.location` will be a simple quadruple (e.g., `[1,2,3,4]`). When there are 2 or more views then `window.location` will be a list of quadruples (e.g., `[[1,2,3,4], [5,6,7,8]]`). I chose to implement a list of quadruples rather than a dict of quadruples because the View UID will in most cases be auto-generated and not be meaningful for the end user. Instead the order of views should be easy to understand, i.e., the first view's location is `widget.location[0]`.